lib/hasPermissions.ts

```diff
@@ -232,13 +225,21 @@ export class HasPermissionsDirectiveVisitor<
         }
       }
 
-      const enhancedInfo = {
-        ...info,
-        missingPermissions,
-      };
+      // eslint-disable-next-line no-param-reassign
+      resolverInfo.missingPermissions = missingPermissions;
 
-      return resolve.apply(this, [obj, args, context, enhancedInfo]);
+      if (missingPermissions) {
+        // TODO Luis: This will make the current value turns into 'null' since
+        // this couldn't be validated. This isn't the desired behaviour to this
+        // specific directive. Thing in a way to keep the original value when the
+        // validation fails
```

Comment:
IMO in case the policy is resolver, we should not nullify the field. Leave it as is and let the resolver decide what to do. This is a corner case for this directive.